src/mapillary.py

```diff
@@ -0,0 +1,145 @@
+        self.basepath.mkdir(parents=True, exist_ok=True)
+        self.client = Session()
+        self.client.mount("https://", HTTPAdapter(max_retries=3))
+
+    def get_image_from_coordinates(self, point: Point) -> dict:
+        longitude, latitude = point.x, point.y
+        log.info("Get Image From Coordinates: %s, %s", latitude, longitude)
+        try:
+            response = self.client.get(
+                self.url,
+                params={
+                    "access_token": self.access_token,
+                    "fields": "id,thumb_original_url",
+                    "is_pano": "true",
+                    "bbox": self._bounds(latitude, longitude),
+                },
+            )
+            response.raise_for_status()
+        except RequestException as e:
+            log.error(e)
+            return {
+                "latitude": latitude,
+                "longitude": longitude,
+                "image_id": None,
+                "image_path": None,
+            }
+
+        images = response.json()["data"]
+        log.debug("Successfully Retrieved Image Data: %s", images)
+        if len(images) == 0:
+            log.debug(
+                "No Images in Bounding Box: %s", self._bounds(latitude, longitude)
+            )
+            return {
+                "latitude": latitude,
+                "longitude": longitude,
+                "image_id": None,
+                "image_path": None,
+            }
+
+        image_id = images[0]["id"]
+        image_url = images[0]["thumb_original_url"]
+        image_path = self._download_image(image_url, image_id)
+
+        return {
+            "latitude": latitude,
+            "longitude": longitude,
+            "image_id": image_id,
+            "image_path": str(image_path),
+        }
+
+    def _bounds(self, latitude, longitude) -> str:
+        left = longitude - 10 / 111_111
+        bottom = latitude - 10 / 111_111
+        right = longitude + 10 / 111_111
+        top = latitude + 10 / 111_111
+        return f"{left},{bottom},{right},{top}"
+
+    def _download_image(self, image_url, image_id) -> Optional[Path]:
+        log.info("Downloading Image: %s", image_id)
+        try:
+            response = self.client.get(image_url, stream=True)
+            response.raise_for_status()
+        except RequestException as e:
+            log.error(e)
+            return None
+        image_content = response.content
+        log.debug("Successfully Retrieved Image: %s", image_id)
+        image_path = Path(self.basepath, f"{image_id}.jpeg")
+        log.debug("Writing Image To: %s", image_path)
+
+        if not image_path.is_file():
+            with open(image_path, "wb") as img:
+                img.write(image_content)
+            log.debug("Successfully Wrote Image: %s", image_path)
+
+        return image_path
+
+
+@app.command()
+def main(
+    client_token: Annotated[
+        str,
+        Argument(help="Mapillary Client Token from Developer Portal"),
```

Comment:
Instead of passing in the Mapillary Client Token as an input, I think the implementation that you had in #12 with an environment variable and `load_dotenv` was better. Since this is a secret, it's better to have it be an environment variable, and supporting a `.env` file is nice because users can just set it in a secret file and not need to worry about looking it up again and copy-pasting. Requiring people pass it in at the command line means it will be in people's shell history, etc.